RISK_MATRIX.md

```diff
@@ -0,0 +1,61 @@
+# Risk consideration
```

Comment:
If you make this file `.mdx` and add the elastic-docs front-matter, we could add this doc to the developer guide.